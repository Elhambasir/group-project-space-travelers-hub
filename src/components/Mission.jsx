import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchMissions } from '../redux/mission/missionSlice';

const Mission = () => {
  const dispatch = useDispatch();
  const missions = useSelector((store) => store.missions);
  useEffect(() => {
    if (!missions.length) dispatch(fetchMissions());
  }, [dispatch]); // eslint-disable-line
  return (
    <div>hello world</div>
  );
};

export default Mission;
