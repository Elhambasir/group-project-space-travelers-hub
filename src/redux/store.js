// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import MissionReducer from './mission/missionSlice';

// eslint-disable-next-line import/prefer-default-export
export const store = configureStore({
  reducer: {
    missions: MissionReducer,
  },
});
