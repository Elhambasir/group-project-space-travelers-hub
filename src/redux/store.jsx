// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './Rockets/rocketsSlice';
import MissionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: MissionReducer,
  },
});

export default store;
