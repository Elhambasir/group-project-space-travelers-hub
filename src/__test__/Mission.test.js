import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';
import store from '../redux/store';

describe('Component render testing', () => {
  it('Mission page renders correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Mission />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
