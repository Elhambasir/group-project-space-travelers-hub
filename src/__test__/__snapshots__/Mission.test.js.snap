// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render testing Mission page renders correctly 1`] = `<DocumentFragment />`;
